python_modules/libraries/dagster-k8s/dagster_k8s/component.py

```diff
@@ -0,0 +1,62 @@
+from collections.abc import Mapping, Sequence
+from dataclasses import dataclass
+from functools import cached_property
+from typing import Any, Optional, Union
+
+from dagster import AssetExecutionContext, AssetsDefinition, Definitions, multi_asset
+from dagster.components import Component, ComponentLoadContext, Resolvable, ResolvedAssetSpec
+
+from dagster_k8s.pipes import PipesK8sClient, build_pod_body
+
+
+@dataclass
+class PipesK8sSpec(Resolvable):
```

Comment:
open to better ideas for what to call a unit of k8s pod + the assets it produces . I started with Op which felt worse